tools/engine_tool/lib/src/dart_utils.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi' as ffi show Abi;
+
+import 'package:path/path.dart' as p;
+import 'environment.dart';
+
+String _getAbiSubdirectory(ffi.Abi abi) {
+  switch (abi) {
+    case ffi.Abi.macosArm64:
+      return 'macos-arm64';
+    case ffi.Abi.macosX64:
+      return 'macos-x64';
+    case ffi.Abi.windowsArm64:
+      return 'windows-arm64';
+    case ffi.Abi.windowsX64:
+      return 'windows-x64';
+    case ffi.Abi.androidArm:
+    case ffi.Abi.androidArm64:
+    case ffi.Abi.androidIA32:
+    case ffi.Abi.androidX64:
+    case ffi.Abi.androidRiscv64:
+    case ffi.Abi.fuchsiaArm64:
+    case ffi.Abi.fuchsiaX64:
+    case ffi.Abi.fuchsiaRiscv64:
+    case ffi.Abi.iosArm:
+    case ffi.Abi.iosArm64:
+    case ffi.Abi.iosX64:
+    case ffi.Abi.linuxArm:
+    case ffi.Abi.linuxArm64:
+    case ffi.Abi.linuxIA32:
+    case ffi.Abi.linuxX64:
+    case ffi.Abi.linuxRiscv32:
+    case ffi.Abi.linuxRiscv64:
+    case ffi.Abi.windowsIA32:
+    default:
+      return 'FIXME dart_utils.dart';
+  }
+}
+
+/// Returns the dart-sdk/bin directory.
+String findDartBinDirectory(Environment env) {
```

Comment:
`bin/et{.bat}` invokes the engine tool entrypoint using the prebuilt dart binary under `prebuilts`, so you might also just use the parent directory of  `env.platform.resolvedExecutable` for this.